sys/include/xtimer.h

```diff
@@ -0,0 +1,442 @@
+/*
+ * Copyright (C) 2015 Kaspar Schleiser <kaspar@schleiser.de>
+ *
+ * This file is subject to the terms and conditions of the GNU Lesser
+ * General Public License v2.1. See the file LICENSE in the top level
+ * directory for more details.
+ */
+
+/**
+ * @defgroup  sys_xtimer Timers
+ * @ingroup   sys
+ * @brief     Provides a high level timer module to register
+ *            timers, get current system time, and let a thread sleep for
+ *            a certain amount of time.
+ *
+ * The implementation takes one low-level timer that is supposed to run at 1MHz
+ * speed and multiplexes it.
+ *
+ * Insertion and removal of timers has O(n) complexity with (n) being the
+ * number of active timers.  The reason for this is that multiplexing is
```

Comment:
To see it from a user's perspective: I have the concrete use case that I want to run a network stack, e.g. *gnrc* on top of *IEEE 802.15.4e* (*TSCH* = Time Slotted Channel Hopping) which is basically a TDMA MAC. Most likely the *TSCH* implementation (currently this is taken from *OpenWSN*) will use `periph/timers`. But if *gnrc* is using *xtimers*, I will run into trouble if they are not deterministic. E.g. my application may know the node has a sending timeslot 'A' in `x µs`. If the (rest of the) stack is deterministic (take this for granted right now) I can calculate if I can use 'A' for sending out a, let's say, UDP packet or not. If the stack is not deterministic because of timers being set in a non-deterministic manner, I cannot know this.